cmd/vic-machine/converter/converter.go

```diff
@@ -0,0 +1,508 @@
+	if !ok {
+		panic(fmt.Sprintf(src.Type().String() + " is not IP"))
+	}
+	if ip.IsUnspecifiedIP(ipAddr) {
+		return nil
+	}
+	v := ipAddr.String()
+
+	log.Debugf("%s=%s", prefix, v)
+	addValue(dest, prefix, v)
+	return nil
+}
+
+func convertIPRange(src reflect.Value, prefix string, tags reflect.StructTag, dest map[string][]string) error {
+	defer trace.End(trace.Begin(fmt.Sprintf("prefix: %s, src: %s", prefix, src.String())))
+
+	if prefix == "" {
+		return nil
+	}
+	if tags.Get(cmdTag) == "" {
+		return nil
+	}
+
+	ipRange, ok := src.Interface().(ip.Range)
+	if !ok {
+		panic(fmt.Sprintf(src.Type().String() + " is not ip range"))
+	}
+	v := ipRange.String()
+	if v == "" {
+		return nil
+	}
+
+	log.Debugf("%s=%s", prefix, v)
+	addValue(dest, prefix, v)
+	return nil
+}
+
+func convertString(src reflect.Value, prefix string, tags reflect.StructTag, dest map[string][]string) error {
+	defer trace.End(trace.Begin(fmt.Sprintf("prefix: %s, src: %s", prefix, src.String())))
+
+	if prefix == "" {
+		return nil
+	}
+	if tags.Get(cmdTag) == "" {
+		return nil
+	}
+
+	v := src.String()
+	if v == "" {
+		return nil
+	}
+
+	log.Debugf("%s=%s", prefix, v)
+
+	addValue(dest, prefix, v)
+	return nil
+}
+
+func convertPrimitive(src reflect.Value, prefix string, tags reflect.StructTag, dest map[string][]string) error {
+	defer trace.End(trace.Begin(fmt.Sprintf("prefix: %s, src: %s", prefix, src.String())))
+
+	if prefix == "" {
+		return nil
+	}
+	if tags.Get(cmdTag) == "" {
+		return nil
+	}
+
+	v := ""
+	switch src.Kind() {
+	case reflect.Int, reflect.Int8, reflect.Int16, reflect.Int32, reflect.Int64:
+		if src.Int() == 0 {
+			return nil
+		}
+		v = strconv.FormatInt(src.Int(), 10)
+	case reflect.Uint, reflect.Uint8, reflect.Uint16, reflect.Uint32, reflect.Uint64:
+		if src.Uint() == 0 {
+			return nil
+		}
+		v = strconv.FormatUint(src.Uint(), 10)
+	case reflect.Bool:
+		if !src.Bool() {
+			return nil
+		}
+		v = strconv.FormatBool(src.Bool())
+	case reflect.Float32, reflect.Float64:
+		if src.Float() == 0 {
+			return nil
+		}
+		v = strconv.FormatFloat(src.Float(), 'E', -1, 64)
+	default:
+		panic(fmt.Sprintf("%s is not supported type", src.Kind()))
+	}
+	log.Debugf("%s=%s", prefix, v)
+
+	addValue(dest, prefix, v)
+	return nil
+}
+
+// convertNetwork will merge destination and gateway to one option with format: 192.168.3.0/16,192.168.128.0/16:192.168.2.1
+// after that, convertStruct is called for left conversion
+func convertNetwork(src reflect.Value, prefix string, tags reflect.StructTag, dest map[string][]string) error {
+	defer trace.End(trace.Begin(fmt.Sprintf("prefix: %s, src: %s", prefix, src.String())))
+
+	if prefix == "" {
+		return nil
+	}
+	if tags.Get(cmdTag) == "" {
+		return nil
+	}
+
+	network, ok := src.Interface().(data.NetworkConfig)
+	if !ok {
+		panic(fmt.Sprintf(src.Type().String() + " is not NetworkConfig"))
+	}
+	if !network.IsSet() {
+		return nil
+	}
+
+	if len(network.Destinations) > 0 || !ip.Empty(network.Gateway) {
```

Comment:
It would be nice to add a comment explaining this `if` block since this code is a bit involved.